src/sentry/api/endpoints/organization_member_invite/index.py

```diff
@@ -0,0 +1,283 @@
+            organization_id=self.context["organization"].id,
+            is_verified=False,
+        )
+        member_queryset = OrganizationMember.objects.filter(
+            Q(user_id__in=[u.id for u in users]),
+            organization=self.context["organization"],
+        )
+
+        if member_queryset.exists():
+            raise MemberConflictValidationError("The user %s is already a member" % email)
+
+        # check for existing invites
+        invite_queryset = OrganizationMemberInvite.objects.filter(
+            Q(email=email),
+            organization=self.context["organization"],
+        )
+        if invite_queryset.filter(invite_status=InviteStatus.APPROVED.value).exists():
+            raise MemberConflictValidationError("The user %s has already been invited" % email)
+
+        if not self.context.get("allow_existing_invite_request"):
+            if invite_queryset.filter(
+                Q(invite_status=InviteStatus.REQUESTED_TO_BE_INVITED.value)
+                | Q(invite_status=InviteStatus.REQUESTED_TO_JOIN.value)
+            ).exists():
+                raise MemberConflictValidationError(
+                    "There is an existing invite request for %s" % email
+                )
+
+        return email
+
+    def validate_orgRole(self, role):
+        if role == "billing" and features.has(
+            "organizations:invite-billing", self.context["organization"]
+        ):
+            return role
+        role_obj = next((r for r in self.context["allowed_roles"] if r.id == role), None)
+        if role_obj is None:
+            raise serializers.ValidationError(
+                "You do not have permission to set that org-level role"
+            )
+        if not self.context.get("allow_retired_roles", True) and role_obj.is_retired:
+            raise serializers.ValidationError(
+                f"The role '{role}' is deprecated and may no longer be assigned."
+            )
+        return role
+
+    def validate_teams(self, teams):
+        valid_teams = list(
+            Team.objects.filter(
+                organization=self.context["organization"], status=TeamStatus.ACTIVE, slug__in=teams
+            )
+        )
+
+        if len(valid_teams) != len(teams):
+            raise serializers.ValidationError("Invalid teams")
+
+        return valid_teams
+
+
+@region_silo_endpoint
+@extend_schema(tags=["Organizations"])
+class OrganizationMemberInviteIndexEndpoint(OrganizationEndpoint):
+    # TODO (mifu67): make these PUBLIC once ready
+    publish_status = {
+        "GET": ApiPublishStatus.EXPERIMENTAL,
+        "POST": ApiPublishStatus.EXPERIMENTAL,
+    }
+    permission_classes = (MemberAndStaffPermission,)
+    owner = ApiOwner.ENTERPRISE
+
+    def get(self, request: Request, organization) -> Response:
+        """
+        List all organization member invites.
+        """
+        queryset = OrganizationMemberInvite.objects.filter(organization=organization).order_by(
```

Comment:
when this ordering happens, which invite objects are shown first? it would be nice to see the join requests first